tests/_test_base.bash

```diff
@@ -11,7 +11,10 @@ source "$SECRET_PROJECT_ROOT/src/_utils/_git_secret_tools.sh"
 # Constants:
 FIXTURES_DIR="$BATS_TEST_DIRNAME/fixtures"
 
```

Comment:
Can we add comment(s) here describing where we expect the /run and /gpg directories to be created? 
This will help users understand effect.

Also can the comment describe BATS_TMPDIR and SECRET_PROJECT_ROOT, and how/where it expects those variables to get set?

Reply:
Sure, I can add those. I'm thinking that maybe I should look at the gpg-agent issue before I submit the final solution to the directory issue. The problem I see is that launching and killing the gpg-agent, unless we can guarantee that we kill the right agent, is bad. Therefore we need a persistent directory to act as homedir. Maybe we should move the requirement of having a running gpg-agent to outside of each separate test?

Reply:
@Mohjive I think it's fine to merge this branch (when we're happy with it) before fixing the `gpg-agent` issue, which only effects `make test` and has been in the code base for quite some time.

But I agree, we should resolve the `gpg-agent` issue soon.

Reply:
These changes also only affect `make test` afaik. 

The reason I thought about fixing the gpg-agent issue before I could make this PR final is because I'm unsure how to solve the persistent gpg homedir (if there should be one), and deciding whether there should be a persistent homedir or not decides how and which directories should be created imo.

Reply:
I'd say we should move the tests towards using a clean environment/new homedir for each test. It's the easiest to get right and the most similar to the current code base. 

I'm not against the persistent homedir idea, but I suspect it's a bigger PR, and I think it might be harder to make logically correct with all the versions of gpg out in the wild.

I'm happy to let you decide when your PR should get merged, too :]